ui/src/shared/components/Gauge.tsx

```diff
@@ -349,10 +348,18 @@ class Gauge extends Component<Props> {
 
   private drawNeedle = (ctx, radius, minValue, maxValue) => {
     const {gaugePosition} = this.props
-    const {degree, needleColor0, needleColor1} = GAUGE_SPECS
+    const {degree, needleColor0, needleColor1, overflowDelta} = GAUGE_SPECS
     const arcDistance = Math.PI * 1.5
 
-    const needleRotation = (gaugePosition - minValue) / (maxValue - minValue)
+    let needleRotation: number
+
+    if (gaugePosition <= minValue) {
```

Comment:
This is awesome! how do you feel about a function like `getNeedleRotation` and sticking that logic there?